delegates.rb

```diff
@@ -83,93 +84,96 @@ def redirect(options = {})
   def authorize(options = {})
     logger = Java::edu.illinois.library.cantaloupe.script.Logger
    
-    u_file_access = ['10.128.99.55','10.128.1.167','10.224.6.10','10.128.99.167','10.128.98.50','10.224.6.26','10.224.6.35','172.16.1.94', '66.234.38.35']
+    # full_res_file_access = ['10.128.99.55','10.128.1.167','10.224.6.10','10.128.99.167','10.128.98.50','10.224.6.26','10.224.6.35','172.16.1.94', '66.234.38.35']
     #'65.88.88.115'
     remote_ip = context['request_headers']['X-Forwarded-For']
     logger.debug("CONTEXT HASH: #{context}")
+    logger.debug("REQUEST HEADERS: #{context['request_headers']}")
+    logger.debug("CLIENT_IP: #{context['client_ip']}")
     logger.debug("IP ADDRESS: #{remote_ip}")
     logger.debug("REQUEST URI: #{context['request_uri']}")
-    # set type to variable since it will be referenced more frequently in future work
-    type = context['request_uri'].split('=')[1]
-    logger.debug("type: #{type}")
-    if context['request_uri'] =~ /ufile=true/
-      logger.debug("UFILE ACCESS")
-      if u_file_access.include?(remote_ip) || remote_ip =~ /^63.147.60./
-        true
+    type = derivative_type(context['resulting_size'])
+    logger.debug("TYPE: #{type}")
+    logger.debug("NON_UFILE ACCESS")
+    rights = get_rights(context['identifier'], context['client_ip'])
+    is_not_restricted_for_ip = returns_rights?(rights) && is_not_restricted?(rights, type) #return true if not restricted
+    # if type == "full_res"
+    #   logger.debug("FULL RES FILE ACCESS")
+    #   if full_res_file_access.include?(remote_ip) || remote_ip =~ /^63.147.60./
+    #     true
+    #   else
+    #     false
+    #   end
+    # else
+      # logger.debug("NON_UFILE ACCESS")
+      # rights = get_rights(context['identifier'], context['client_ip'])
+      # is_not_restricted_for_ip = returns_rights?(rights) && is_not_restricted?(rights, type) #return true if not restricted
+    # end
+  end
+
+  def derivative_type(size)
+    width = size["width"]
+    height = size["height"]
+    case
+      when (width <= 100) || (width <= 100) 

```

Comment:
did you mean height?

Reply:
in any case I think we want the longest side, so I would probably choose the longest side first (longest_side = width > height ? width : height), and just use that. Otherwise we could invite some extreme derivatives of insane length on one side by small on the other.

Reply:
I added it as a failsafe incase there is an image that is not in landscape orientation (wasn't sure if that was the standard or not)